vslam/feature/sift_feature.py

```diff
@@ -14,17 +14,17 @@ def __init__(self, **kwargs) -> None:
         super().__init__()
         self.sift = cv2.SIFT_create(**kwargs)
 
-    def detect(self, img: np.ndarray, mask: np.ndarray = None) -> list:
+    def detect(self, img: np.ndarray, mask: np.ndarray = None) -> np.ndarray:
```

Comment:
if it's a list, type annotation must be List with (include from typing import List)